backpack/core/derivatives/lstm.py

```diff
@@ -0,0 +1,358 @@
+"""Partial derivatives for nn.LSTM."""
+from typing import Tuple
+
+from torch import Tensor, allclose, cat, einsum, sigmoid, tanh, zeros
+from torch.nn import LSTM
+
+from backpack.core.derivatives.basederivatives import BaseParameterDerivatives
+
+
+class LSTMDerivatives(BaseParameterDerivatives):
+    """Partial derivatives for nn.LSTM layer.
+
+    Index conventions:
+    ------------------
+    * t: Sequence dimension
+    * v: Free dimension
+    * n: Batch dimension
+    * h: Output dimension
+    * i: Input dimension
+    """
+
+    @staticmethod
+    def _check_parameters(module: LSTM) -> None:
+        """Check the parameters of module.
+
+        Args:
+            module: module which to check
+
+        Raises:
+            NotImplementedError: If any parameter of module does not match expectation
+        """
+        if module.num_layers > 1:
+            raise NotImplementedError("only num_layers = 1 is supported")
+        if module.bias is not True:
+            raise NotImplementedError("only bias = True is supported")
+        if module.batch_first is not False:
+            raise NotImplementedError("only batch_first = False is supported")
+        if not module.dropout == 0:
+            raise NotImplementedError("only dropout = 0 is supported")
+        if module.bidirectional is not False:
+            raise NotImplementedError("only bidirectional = False is supported")
+        if not module.proj_size == 0:
+            raise NotImplementedError("only proj_size = 0 is supported")
+
+    @staticmethod
+    def _forward_pass(
+        module: LSTM, mat: Tensor
+    ) -> Tuple[Tensor, Tensor, Tensor, Tensor]:
+        T: int = mat.shape[1]
+        N: int = mat.shape[2]
+        H: int = module.hidden_size
+        H0: int = 0 * H
+        H1: int = 1 * H
+        H2: int = 2 * H
+        H3: int = 3 * H
+        H4: int = 4 * H
+        # forward pass and save i, f, g, o, c, c_tanh-> ifgo, c, c_tanh
+        ifgo: Tensor = zeros(T, N, 4 * H, device=mat.device)
+        c: Tensor = zeros(T, N, H, device=mat.device)
+        c_tanh: Tensor = zeros(T, N, H, device=mat.device)
+        h: Tensor = zeros(T, N, H, device=mat.device)
+        for t in range(T):
+            if t == 0:
+                ifgo[t] = (
+                    einsum("hi, ni -> nh", module.weight_ih_l0, module.input0[t])
+                    + module.bias_ih_l0
+                    + module.bias_hh_l0
+                )
+            else:
+                ifgo[t] = (
+                    einsum("hi, ni -> nh", module.weight_ih_l0, module.input0[t])
+                    + module.bias_ih_l0
+                    + einsum("hg, ng -> nh", module.weight_hh_l0, module.output[t - 1])
+                    + module.bias_hh_l0
+                )
+            ifgo[t, :, H0:H1] = sigmoid(ifgo[t, :, H0:H1])
```

Comment:
I just tested that this should work on a simple example: `ifgo[t, :, H0:H1].sigmoid_()`

Reply:
Does it add benefit here?
I don't know for sure but I would hope that the PyTorch implementation is efficient.

Reply:
just reads shorter, but I agree that it can be left as is.